[17.0][MIG] stock_secondary_unit: Migration to 17.0

rov-adhoc opened the pull request.

sergio-teruel committed:
[ADD] stock_secondary_unit: New module for getting stock in a secondary unit

sergio-teruel committed:
[11.0][FIX] stock_secondary_unit: Remove procurement_rule move. Fix dependencies

kittiu committed:
[12.0][MIG] stock_secondary_unit

sergio-teruel committed:
[FIX] stock_secondary_unit: Fix key error in vals

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (15 of 15 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_secondary_unit/zh_CN/

digitalsatori committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (15 of 15 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_secondary_unit/zh_CN/

enriquemartin committed:
[12.0][FIX] stock_secondary_unit: move line create round
On move line creation, the precision of secondary qty rounding is wrong; the unit of measure factor is being used instead of rounding.

ps-tubtim committed:
[IMP] stock_secondary_unit: black, isort

ps-tubtim committed:
[MIG] stock_secondary_unit: Migration to 13.0

sergio-teruel committed:
[IMP] stock_secondary_unit: Display info only for inventory users

sergio-teruel committed:
[IMP] stock_secondary_unit: Remove unnecessary code. Always compute as field.
TT26606

sergio-teruel committed:
[FIX] stock_secondary_unit: Incorrect secondary unit link in product variants view on common form. Create related template secondary unit field.
TT27057

sergio-teruel committed:
[IMP] stock_secondary_unit: Use secondary units directly in pickings
[IMP] stock_secondary_unit: Use product_secondary_unit mixin model from product-attribute repository.

sergio-teruel committed:
[FIX] stock_secondary_unit: Allow use secondary units in form view for mobile views
TT30538

alan196 committed:
[IMP] stock_secondary_unit: black, isort, prettier

alan196 committed:
[MIG] stock_secondary_unit: migration to V14.

CarlosRoca13 committed:
[MIG] stock_secondary_unit: Migration to 15.0

Commit:
[UPD] Update stock_secondary_unit.pot

OCA-git-bot committed:
[UPD] README.rst

Commit:
[UPD] Update stock_secondary_unit.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_secondary_unit/

sergio-teruel committed:
[IMP] stock_secondary_unit: Compatibility with product variants
TT40323

sergio-teruel committed:
[FIX] stock_secondary_unit: Remove _merge_moves_fields method. Now Odoo compute the change

sergio-teruel committed:
[FIX] stock_secondary_unit: Propagate secondary_uom_id for new extra moves

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (14 of 14 strings)

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_secondary_unit/es/

n17 committed:
Added translation using Weblate (Finnish)

n17 committed:
Translated using Weblate (Finnish)

Currently translated at 92.8% (13 of 14 strings)

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_secondary_unit/fi/

OCA-git-bot committed:
[UPD] README.rst

josep-tecnativa committed:
[IMP] stock_secondary_unit: test performance improvement
- Include context keys for avoiding mail operations overhead.

OCA-git-bot committed:
[BOT] post-merge updates

sergio-teruel committed:
[FIX] stock_secondary_unit: The second qty is computed when a sml is created. Second qty compute the qty_done field.
The second uom not locked for non draft move lines

Commit:
[UPD] Update stock_secondary_unit.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_secondary_unit/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_secondary_unit/it/

sergio-teruel committed:
[IMP] stock_secondary_unit: Make readonly secondary qty field in some states

OCA-git-bot committed:
[BOT] post-merge updates

lk-eska committed:
[MIG] stock_secondary_unit: Migration to 16.0

pilarvargas-tecnativa committed:
[IMP] stock_secondary_unit: make the secondary unit and secondary quantity fields optional

TT49686

rov-adhoc commented:
@rousseldenis  Hey! I’d like to ask if there’s anything missing for it to be approved. Thanks in advance! Best regards!

rousseldenis commented:
/ocabot migration stock_secondary_unit

rousseldenis commented:
> @rousseldenis Hey! I’d like to ask if there’s anything missing for it to be approved. Thanks in advance! Best regards!

Reviews :-)

rousseldenis commented on stock_secondary_unit/models/stock_move.py:
```diff
@@ -12,16 +12,36 @@ class StockMove(models.Model):
-    def onchange_product_uom_for_secondary(self):
-        self._onchange_helper_product_uom_for_secondary()
+    @api.depends("product_id")
+    def _compute_product_uom(self):
```
@rov-adhoc Could you try to remove completely the compute as it should be computed by the mixin.

See: https://github.com/OCA/product-attribute/pull/1560/files#diff-4111726e152eb08dc8a785703aa8b37a47652a6d7abb0d13ab97b41da579c9eaR18

rov-adhoc replied:
@rousseldenis You're right!
Thanks

Commit:
[UPD] Update stock_secondary_unit.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_secondary_unit/

pilarvargas-tecnativa commented:
Please include https://github.com/OCA/stock-logistics-warehouse/pull/2080

pilarvargas-tecnativa commented:
Sorry but I have seen that there are 2 other PRs for this migration, please indicate if this one overwrites them to be closed.

OCA-git-bot committed:
[BOT] post-merge updates

rov-adhoc commented:
@pilarvargas-tecnativa Hey! I just added it!

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: stock-logistics-warehouse-16.0/stock-logistics-warehouse-16.0-stock_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-16-0/stock-logistics-warehouse-16-0-stock_secondary_unit/it/

rov-adhoc commented:
@fredericgrall Hey! Would you mind closing your PR and reviewing this one when you have a moment? I’d really appreciate it. Thanks in advance!

rousseldenis reviewed (approved):
LGTM

rov-adhoc commented:
@pilarvargas-tecnativa Hi!! I would like to ask you if you can review this PR.
Thanks in advance

ced-adhoc committed:
[16.0][FIX] stock_secondary_unit: use @api.model_create_multi in create method

OCA-git-bot committed:
[BOT] post-merge updates

pilarvargas-tecnativa commented on stock_secondary_unit/views/stock_picking_views.xml:
```diff
@@ -6,15 +6,15 @@
                     name="secondary_uom_qty"
-                    attrs="{'column_invisible': [('parent.immediate_transfer', '=', True)], 'readonly': ['|', ('is_initial_demand_editable', '=', False), '&amp;', '&amp;', ('show_operations', '=', True), ('is_locked', '=', True), ('is_initial_demand_editable', '=', False)]}"
+                    groups="uom.group_uom"
+                    readonly="not is_initial_demand_editable"
```
Wouldn't the column_invisible attribute and the rest of the readonly conditions be missing?

pilarvargas-tecnativa commented on stock_secondary_unit/views/stock_picking_views.xml:
```diff
@@ -23,27 +23,11 @@
                     options="{'no_create': True}"
-                    attrs="{'column_invisible': [('parent.immediate_transfer', '=', True)], 'readonly': ['|', ('is_initial_demand_editable', '=', False), '&amp;', '&amp;', ('show_operations', '=', True), ('is_locked', '=', True), ('is_initial_demand_editable', '=', False)]}"
+                    groups="uom.group_uom"
+                    readonly="not is_initial_demand_editable"
```
Same here

rov-adhoc replied:
@pilarvargas-tecnativa Hi! You are correct about the readonly! I just corrected it. Regarding column_invisible, the immediate_transfer field no longer exists in version 17 of the stock.picking model.

lav-adhoc reviewed (approved):
LGTM

sergio-teruel committed:
[FIX] stock_secondary_unit: Error when user try to select a secondary uom without product in sale order line
TT51683

rov-adhoc commented:
@rousseldenis HI!! I would like to ask you if is something missing out? Regards

pilarvargas-tecnativa commented:
> @rousseldenis HI!! I would like to ask you if is something missing out? Regards

Hello, it remains to answer my previous question. Regards

![image](https://github.com/user-attachments/assets/08285b06-f654-40f7-856e-6148c8eaea9e)

OCA-git-bot committed:
[BOT] post-merge updates

rov-adhoc commented:
@pilarvargas-tecnativa Hi!
Thank you for letting me know! I hadn't seen that message! I've added new changes.

pilarvargas-tecnativa commented on stock_secondary_unit/views/stock_picking_views.xml:
```diff
@@ -23,27 +25,11 @@
-                    options="{'no_create': True}"
-                    attrs="{'invisible': [('parent.immediate_transfer', '=', True)], 'readonly': [('is_initial_demand_editable', '=', False)]}"
-                />
-            </xpath>
```
https://github.com/OCA/stock-logistics-warehouse/commit/9bead1609195f4eb183826b619a387e5e81f8889#diff-7b894bb25455e77e828781f63f3c91bd533fc3fc43770d401ba5b29913c6b972R27

Is this no longer necessary?

rov-adhoc replied:
Hi! I the field immediate_transfer is no longer exits in stock_picking

pilarvargas-tecnativa replied:
I am not referring to the conditions for the visibility of the fields, those fields were added for a reason, they are no longer needed?

Commit: https://github.com/OCA/stock-logistics-warehouse/commit/9bead1609195f4eb183826b619a387e5e81f8889#diff-7b894bb25455e77e828781f63f3c91bd533fc3fc43770d401ba5b29913c6b972R27

![image](https://github.com/user-attachments/assets/436c09d7-7c14-4809-910d-6acbac414899)

rov-adhoc replied:
@pilarvargas-tecnativa Thank you for clarifying. I have added them again. I hope they are correct now.

sergio-teruel commented:
Hi... Can you add the 4061a97e1212248e9c4855c55894a4f3712dd8a0 commit from 16?

rov-adhoc commented:
@sergio-teruel added it

pilarvargas-tecnativa commented on stock_secondary_unit/views/stock_picking_views.xml:
```diff
@@ -23,17 +25,15 @@
-                expr="//field[@name='move_ids_without_package']/form//field[@name='product_uom_qty']"
-                position="before"
-            >
+            <xpath expr="//field[@name='product_uom_qty']" position="before">
```
That xpath is not correct, it duplicates the fields in the tree of the form where you have already added them before.

We have to see how the mobile view is built in this version, if there are changes and if you need to add the fields to make it work exactly the same as the previous version.

peluko00 reviewed (approved):
LGTM

pilarvargas-tecnativa reviewed (changes requested):
https://github.com/OCA/stock-logistics-warehouse/pull/2121#discussion_r1843723939

juancarlosonate-tecnativa commented:
@rov-adhoc Could you take a look at https://github.com/OCA/stock-logistics-warehouse/pull/2121#discussion_r1843723939 and check if everything is correct, or should I open a new PR?

rov-adhoc committed:
[IMP] stock_secondary_unit: pre-commit execution

rov-adhoc committed:
[MIG] stock_secondary_unit: Migration to 17.0

rov-adhoc commented:
> > @rousseldenis HI!! I would like to ask you if is something missing out? Regards
> 
> Hello, it remains to answer my previous question. Regards
> 
> ![image](https://private-user-images.githubusercontent.com/118818446/385351700-08285b06-f654-40f7-856e-6148c8eaea9e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzMzkyNDAsIm5iZiI6MTczNjMzODk0MCwicGF0aCI6Ii8xMTg4MTg0NDYvMzg1MzUxNzAwLTA4Mjg1YjA2LWY2NTQtNDBmNy04NTZlLTYxNDhjOGVhZWE5ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwOFQxMjIyMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zOWExNDM1M2JhMWVlYTEyMDA3MDQ3NzQ4ZmFiYzEyN2JkMjZhNmU0ZjBlZjkxYTBjYzY4MjM1NGM1NzM5YzI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Y67q2hvQ4l4AAr1SQod9Yc65fP6rL1gPiqbRpjdEAsI)

I just corrected it

rov-adhoc commented:
> @rov-adhoc Could you take a look at [#2121 (comment)](https://github.com/OCA/stock-logistics-warehouse/pull/2121#discussion_r1843723939) and check if everything is correct, or should I open a new PR?

I just added it

pilarvargas-tecnativa commented:
> I just added it

Hi @rov-adhoc ,

The comment to which @juancarlosonate-tecnativa  refers is the one about the xpath error: https://github.com/OCA/stock-logistics-warehouse/pull/2121/files#r1843723939

You have marked it as resolved but the error persists. With an incorrect xpath the fields are duplicated and are not visible in the mobile view, which was the purpose of the commit I mentioned when you removed this part of the code. When you add this code again, which has its purpose, you have to adapt it so that it can be seen correctly in mobile views.

![image](https://github.com/user-attachments/assets/e9bdf4ac-7ddb-4e45-963a-bc364ec640ca)

![image](https://github.com/user-attachments/assets/f7157c04-d673-4328-a72a-3b7af58dbc6a)

This is v16 and is how it should look in mobile view:

![image](https://github.com/user-attachments/assets/d4e5714d-599c-4607-b208-f9c2e10445bc)

rov-adhoc commented:
> > I just added it
> 
> Hi @rov-adhoc ,
> 
> The comment to which @juancarlosonate-tecnativa refers is the one about the xpath error: https://github.com/OCA/stock-logistics-warehouse/pull/2121/files#r1843723939
> 
> You have marked it as resolved but the error persists. With an incorrect xpath the fields are duplicated and are not visible in the mobile view, which was the purpose of the commit I mentioned when you removed this part of the code. When you add this code again, which has its purpose, you have to adapt it so that it can be seen correctly in mobile views.
> 
> ![image](https://private-user-images.githubusercontent.com/118818446/401147498-e9bdf4ac-7ddb-4e45-963a-bc364ec640ca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzNDk2ODAsIm5iZiI6MTczNjM0OTM4MCwicGF0aCI6Ii8xMTg4MTg0NDYvNDAxMTQ3NDk4LWU5YmRmNGFjLTdkZGItNGU0NS05NjNhLWJjMzY0ZWM2NDBjYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwOFQxNTE2MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNDA2NjAyNzUwNDM0NWI5ODJiNGZjMjIwMDhhMTA0MTNiODdmNGU4ODgxMGI1MGVjYzI5ZTRlNjcyOGIyOGY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ar9cFPNJl7X-26QC4E85TEvTmq6KPirOAvzjdyIYSlM)
> 
> ![image](https://private-user-images.githubusercontent.com/118818446/401148282-f7157c04-d673-4328-a72a-3b7af58dbc6a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzNDk2ODAsIm5iZiI6MTczNjM0OTM4MCwicGF0aCI6Ii8xMTg4MTg0NDYvNDAxMTQ4MjgyLWY3MTU3YzA0LWQ2NzMtNDMyOC1hNzJhLTNiN2FmNThkYmM2YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwOFQxNTE2MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZmZiZjdjOGY5ZTE0YzJhZjUwOGY3YmUzMWQ3NTRiNTIwNmRiNDVhNzE3N2MyNDJkNmUwYWU5ZmI5MmM5NTU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.qIwxSGEdQogluiHwekahaVpUCCFQbXjcSV3b8c6QG1M)
> 
> This is v16 and is how it should look in mobile view:
> 
> ![image](https://private-user-images.githubusercontent.com/118818446/401148585-d4e5714d-599c-4607-b208-f9c2e10445bc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzNDk2ODAsIm5iZiI6MTczNjM0OTM4MCwicGF0aCI6Ii8xMTg4MTg0NDYvNDAxMTQ4NTg1LWQ0ZTU3MTRkLTU5OWMtNDYwNy1iMjA4LWY5YzJlMTA0NDViYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwOFQxNTE2MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xY2YzOWFkNTU3OGE1N2JiOGMwNjhlMTI5ZDUwNTg2Y2M3MGRkNTkxZWMxYWZmNzU3MTViMjc2MjM0NDQ5YTczJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.II1slf9SLw0q3ohcEFc_mlPpcbxa_ot1OZGWkuiRpPg)

Thank you for providing more details! Now I fully understand what you mean. I’ve made the necessary corrections, and I hope everything is in order now

pilarvargas-tecnativa reviewed (approved):
Good! Thanks!

juancarlosonate-tecnativa reviewed (changes requested):
in the product form, the action_update_quantity_on_hand appears duplicated

https://github.com/OCA/stock-logistics-warehouse/pull/2121/commits/31d1b2b4f008b1b53d94a265080b8f95a500d222#diff-0ed52cda6c2ce263d32165b3d1b7998595886c080423e3e66de63ff19b4a12f1

![image](https://github.com/user-attachments/assets/f515b39b-90fb-4fee-9223-1a8545bc2a25)

juancarlosonate-tecnativa commented on stock_secondary_unit/views/product_views.xml:
```diff
@@ -24,11 +24,14 @@
             ref="stock.product_template_form_view_procurement_button"
         />
         <field name="arch" type="xml">
-            <xpath expr="//button[@name='action_open_quants']" position="after">
+            <xpath
+                expr="//button[@name='action_update_quantity_on_hand']"
+                position="after"
+            >
                 <button
                     type="object"
-                    name="action_open_quants"
-                    attrs="{'invisible':[('type', '!=', 'product')]}"
+                    name="action_update_quantity_on_hand"
+                    invisible="type != 'product'"
                     class="oe_stat_button"
                     icon="fa-building-o"
                     groups="stock.group_stock_user"
```
something must be wrong here because the button shouldnt look like this

rov-adhoc replied:
Can you please check again?

juancarlosonate-tecnativa replied:
i don't see any changes

rov-adhoc replied:
![image](https://github.com/user-attachments/assets/eca79d77-ee9a-43a6-a05d-7a6b43f061b6)

juancarlosonate-tecnativa replied:
Look at v16

![Selección_001](https://github.com/user-attachments/assets/51523dcb-07e4-462e-9379-01d50315c0f9)

rov-adhoc replied:
It appears to be quite similar. Let me check how it behaves when a secondary unit is set for this product
![image](https://github.com/user-attachments/assets/8c73d6c4-d537-4392-8de1-0574ac27b1d1)

juancarlosonate-tecnativa replied:
similar but not the same, the xpath is not correct and neither is the action, i think
![image](https://github.com/user-attachments/assets/c62bac27-7cb1-47eb-b128-8f4dac17b01d)

rov-adhoc replied:
Please check again! Hope now should be ok

pedrobaeza commented:
/ocabot merge nobump
